lib/dag.mli

```diff
@@ -1,16 +1,42 @@
-(* TODOCP *)
-module Pset : sig
-  type t = Geneweb_db.Driver.iper list
+(** DAG construction from URL parameters for genealogical graph display.
+
+    This module builds directed acyclic graphs (DAGs) from URL parameters
+    specifying persons and their Sosa numbers. The resulting DAG can then be
+    rendered as an HTML table via {!Dag2html} and {!DagDisplay}. *)
+
+(** Efficient set of person identifiers. O(log n) operations. *)
+module Iperset : sig
+  type t
   type elt = Geneweb_db.Driver.iper
 
-  val add : 'a -> 'a list -> 'a list
-  val empty : 'a list
-  val elements : 'a list -> 'a list
-  val mem : 'a -> 'a list -> bool
+  val empty : t
+  val add : elt -> t -> t
+  val mem : elt -> t -> bool
+  val elements : t -> elt list
+  val fold : (elt -> 'a -> 'a) -> t -> 'a -> 'a
+end
+
+(** Efficient map from person identifiers. O(log n) operations. *)
+module Ipermap : sig
+  type 'a t
+
+  val empty : 'a t
+  val add : Geneweb_db.Driver.iper -> 'a -> 'a t -> 'a t
+  val find_opt : Geneweb_db.Driver.iper -> 'a t -> 'a option
 end
```

Comment:
It is a duplication of `Driver.Iper.Set` and `Driver.Iper.Map`. Besides, you could use the following syntax:
```ocaml
module Iperset : Set.S with type elt = Geneweb_db.Driver.Iper
```
It is not equivalent with your code as it exposes all the functions of the sets.